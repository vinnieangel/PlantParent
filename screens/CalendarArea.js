import React, { Component } from "react";
import { CalendarList } from "react-native-calendars";
import { StyleSheet, View, Text } from "react-native";
import Ionicons from "react-native-vector-icons/Ionicons";

export default class CalendarScreen extends Component {
  render() {
    return (
      <View style={styles.container}>
        <Text>Calendar Screen</Text>
        <CalendarList
          onVisibleMonthsChange={(months) => {
            console.log("new month", months);
          }}
          pastScrollRange={0}
          futureScrollRange={0}
          scrollEnabled={true}
          showScrollIndicator={true}
        />
        <View style={styles.optionWrapper}>
          <View style={{ flex: 1, flexDirection: "row", alignItems: "center" }}>
            <Ionicons name="ellipse-outline" size={20} color="#51A746" />
            <Text style={{ fontSize: 20, paddingLeft: 10 }}>
              Snake plant needs to be watered.
            </Text>
          </View>
          <View style={{ flex: 1, flexDirection: "row", alignItems: "center" }}>
            <Ionicons name="ellipse-outline" size={20} color="#51A746" />
            <Text style={{ fontSize: 20, paddingLeft: 10 }}>
              Check the soil for snake plant.
            </Text>
          </View>
          <View style={{ flex: 1, flexDirection: "row", alignItems: "center" }}>
            <Ionicons name="ellipse-outline" size={20} color="#51A746" />
            <Text style={{ fontSize: 20, paddingLeft: 10 }}>Task details.</Text>
          </View>
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: "white",
    flex: 1,
    padding: 20,
  },
  subtext: {
    fontSize: 5,
    paddingLeft: 10,
  },
  textWrapper: {
    flex: 1,
    justifyContent: "flex-start",
  },
  optionWrapper: {
    flex: 1,
    flexDirection: "column",
    justifyContent: "flex-start",
  },
});
