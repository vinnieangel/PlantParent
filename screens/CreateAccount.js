import React, { Component } from "react";
import { ScrollView } from "react-native";
import {
  StyleSheet,
  View,
  Image,
  TextInput,
  TouchableOpacity,
  Text,
  Alert,
} from "react-native";

export default class CreateAccount extends Component {
  constructor(props) {
    super(props);
    this.state = {
      username: "",
      password: "",
      preferred_name: "",
      email: "",
    };
  }

  async createAccount() {
    console.log("here");
    await fetch("https://plantparent506.herokuapp.com/users/createAccount", {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
        // 'Content-Type': 'application/x-www-form-urlencoded',
      },
      body: JSON.stringify({
        username: this.state.username,
        password: this.state.password,
        preferred_name: this.state.preferred_name,
        email: this.state.email,
      }),
    }).then(async (res) => {
      let response = await res.json();
      if (response == "Too short") {
        window.alert(
          "Your username is too short! It needs be at least 8 characters long"
        );
      } else if (response == "Duplicate") {
        window.alert("This username already exists. Try another one!");
      } else {
        window.alert("Signed up successfully!");
        //this.props.navigation.navigate("Login");
      }
    });
  }
  checkFields() {
    if (
      this.username == "" ||
      this.password == "" ||
      this.state.preferred_name == "" ||
      this.email == ""
    ) {
      window.alert("All feilds must be filled!");
    } else if (this.state.password.length < 8) {
      window.alert("Password should be at least 8 characters long!");
    } else {
      this.createAccount();
    }
  }
  render() {
    return (
      <View style={styles.container}>
        <View style={styles.accountLink}>
          <TouchableOpacity
            onPress={() => this.props.navigation.navigate("Login")}
          >
            <Text style={{ color: "#5e85bd" }}>Back to Login</Text>
          </TouchableOpacity>
        </View>
        <View style={styles.logoContainer}>
          <Image
            style={styles.logo}
            source={require("./../Images/PlantParent.png")}
          />
        </View>
        <View style={styles.formContainer}>
          <View style={styles.container2}>
            <Text style={styles.fieldsLabel}>Username</Text>
            <TextInput
              placeholder="Enter your username"
              placeholderTextColor="rgba(29, 53, 87, .7)"
              style={styles.input}
              onChangeText={(text) => {
                this.setState({ username: text });
              }}
            />
            <Text style={styles.fieldsLabel}> Password</Text>
            <TextInput
              placeholder="Enter your password"
              placeholderTextColor="rgba(29, 53, 87, .7)"
              secureTextEntry
              style={styles.input}
              onChangeText={(text) => {
                this.setState({ password: text });
              }}
            />
            <Text style={styles.fieldsLabel}> Name</Text>
            <TextInput
              placeholder="Enter your preferred name"
              placeholderTextColor="rgba(29, 53, 87, .7)"
              style={styles.input}
              onChangeText={(text) => {
                this.setState({ preferred_name: text });
              }}
            />
            <Text style={styles.fieldsLabel}> Email</Text>
            <TextInput
              placeholder="Enter email"
              placeholderTextColor="rgba(29, 53, 87, .7)"
              style={styles.input}
              onChangeText={(text) => {
                this.setState({ email: text });
              }}
            />

            <TouchableOpacity style={styles.buttonContainer}>
              <Text
                testID="button1"
                style={styles.buttonText}
                onPress={() => this.checkFields()}
              >
                Create Account
              </Text>
            </TouchableOpacity>
          </View>
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: "white",
    flex: 1,
    padding: 20,
    marginTop: 40,
  },
  container2: {
    padding: 20,
  },
  logoContainer: {
    justifyContent: "center",
    alignItems: "center",
    paddingBottom: 30,
  },
  logo: {
    flex: 1,
    width: 200,
    height: 100,
    resizeMode: "contain",
    alignItems: "center",
  },
  title: {
    color: "#1d3557",
    marginBottom: 10,
    marginTop: 30,
    marginLeft: 100,
    textAlign: "center",
    width: 200,

    justifyContent: "center",
    opacity: 0.9,
    fontSize: 30,
  },
  fieldsLabel: {
    paddingBottom: 10,
  },
  input: {
    height: 50,
    backgroundColor: "white",
    borderWidth: 1,
    borderRadius: 10,
    borderColor: "#30694B",
    marginBottom: 20,
    color: "#383838",
    paddingHorizontal: 10,
  },
  buttonContainer: {
    alignItems: "center",
    padding: 5,
    marginBottom: 10,
    borderRadius: 10,
    // Plant Color
    backgroundColor: "#30694B",
  },
  buttonText: {
    borderColor: "white",
    fontSize: 15,
    color: "white",
    padding: 10,
    borderRadius: 20,
  },
});
