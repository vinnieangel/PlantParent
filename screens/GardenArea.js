import react from "react";
import React, { Component } from "react";
import { SafeAreaView } from "react-native";
import { ScrollView, FlatList } from "react-native";
import { Card } from "react-native-elements";
import { StyleSheet, View, Image, TouchableOpacity, Text } from "react-native";
import { SearchBar } from "react-native-elements";
import { Searchbar } from "react-native-paper";

export default class GardenArea extends React.Component {
 

  userID = this.props.route.params.userID;
  state = {
    plants : [],
    userPlants: [],
    clickedPlantIndex:undefined
  }

  async componentDidMount() {
    await fetch('http://localhost:5000/gardens/get/'+this.userID, {
      method: 'GET',
      headers: {
        'Content-Type': 'application/json'
      }
    }).then(async res => await res.json())
    .then(async res => {
      //I have the result of the garden get: array of userPlant IDs. Now I have to iterate through this
      //and call the userPlants get to get the full object, and put that into my state

      //If I dont have any plants saved
      if(res == "Empty!") {
        throw new Error("Empty!")
      }
      else {
        let userPlants = [];
        for(let userPlantID of res) {
          console.log('USER PLANT ID', userPlantID)
          await fetch('http://localhost:5000/userPlants/getPlant/'+userPlantID, {
            method:'GET',
            header: {
              'Content-Type':'application/json'
            }
          }).then(async result => {
            return await result.json();
          }).then(result => {
            console.log('RES', result)
            userPlants.push(result)
          }).catch(err => console.log("Error: " + err))
        }
        this.setState({userPlants:userPlants})
      }
    }).then(async ()=> {
      //I have all the userPlants in the state. Now I need to do the same thing I did before, except with 
      //userPlants and plants
      
      let plants = [];
      for(let plant of this.state.userPlants) {
        console.log('PLANT ID', plant.plantID)
        await fetch('http://localhost:5000/plants/getPlant/'+plant.plantID, {
          method:'GET',
          header: {
            'Content-Type':'application/json'
          }
        }).then(async result => {
          return await result.json();
        }).then(result => {
          console.log('RES2', result)
          plants.push(result)
        }).catch(err => console.log("Error: " + err))
      }
      this.setState({plants:plants})
    })
    .catch(err => {
      if(err != "Empty"){
        console.log("Error: "+ err)
      }
    });
  }

  render() {
    return (
        <SafeAreaView style={styles.container}>
        <Text style = {styles.title}>
            Welcome to your garden!
        </Text>
        {this.state.plants.length>0 && <FlatList
            data={this.state.userPlants}
            numColumns={2}
            keyExtractor={(item, index) => item._id}
            renderItem={({item, index}) => 
                <View>
                    <TouchableOpacity onPress={()=>{this.props.navigation.navigate('Plant', {
                        userPlant: this.state.userPlants[index],
                        plant: this.state.plants[index],
                    })
                  }}>
                        <Card>
                            <Image style = {styles.plantImage} source={{uri: this.state.plants[index].image}}>
                            </Image>
                        <Card.Title>{item.givenName}</Card.Title>
                        </Card>
                    </TouchableOpacity>
                </View>}
        />}
        {!this.state.plants.length && <Text>
            Your garden is currently empty! Click on the Search Plants tab to get started.
          </Text>}
        </SafeAreaView>
                );
  }
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: "white",
    flex: 1,
    alignItems: "center",
  },
  title: {
    color: "#006400",
    marginBottom: 10,
    marginTop: 40,
    textAlign: "center",
    width: 200,
    justifyContent: "center",
    opacity: 0.9,
    fontSize: 22,
  },
  plantImage: {
    flex: 1,
    height: 100,
    width: 100,
  },
});
