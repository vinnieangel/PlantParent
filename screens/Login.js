import React, { Component } from "react";
import { ScrollView } from "react-native";
import {
  StyleSheet,
  View,
  Image,
  TextInput,
  TouchableOpacity,
  Text,
  Alert,
} from "react-native";
//import symbolicateStackTrace from "react-native/Libraries/Core/Devtools/symbolicateStackTrace";

export default class Login extends Component {
  constructor(props) {
    super(props);
    this.state = {
      username: "",
      password: "",
      type: "",
    };
  }

  async login() {
    fetch("http://localhost:5000/users/login", {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
        // 'Content-Type': 'application/x-www-form-urlencoded',
      },
      body: JSON.stringify({
        username: this.state.username,
        password: this.state.password,
      }),
    })
      .then(async (res) => {
        let response = await res.json();
        if (response == "Not Found") {
          window.alert("Invalid username/password. Try again!");
        } else {
          this.props.navigation.navigate("TabNavigation", {
            userId: response.userId,
          });
        }
      })
      .catch((err) => {
        console.log(err);
      });
  }

  render() {
    return (
      <ScrollView style={styles.container}>
        {/* <View style={styles.container}> */}
        {/* ---------- Logo ---------- */}
        <View style={styles.logoContainer}>
          <Image
            style={styles.logo}
            source={require("./../Images/PlantParent.png")}
          />
        </View>
        <View style={styles.formContainer}>
          <Text style={styles.formTitle}> Username </Text>
          <TextInput
            placeholder="Username or email"
            placeholderTextColor="#b5b5b5"
            style={styles.input}
            autoCapitalize="none"
            onChangeText={(text) => {
              this.setState({ username: text });
            }}
          />

          <Text style={styles.formTitle}> Password </Text>
          <TextInput
            placeholder="Password"
            placeholderTextColor="#b5b5b5"
            secureTextEntry
            style={styles.input}
            onChangeText={(text) => {
              this.setState({ password: text });
            }}
          />

          {/* ------ Buttons ------ */}
          <TouchableOpacity
            testID="button1"
            onPress={() => this.login()}
            style={styles.buttonContainer}
          >
            <Text style={styles.buttonText}>Log In</Text>
          </TouchableOpacity>

          <View style={styles.accountLink}>
            <Text>Don't have an account? </Text>
            <TouchableOpacity
              testID="button2"
              onPress={() => this.props.navigation.navigate("CreateAccount")}
            >
              <Text style={{ color: "#5e85bd" }}>Create Account</Text>
            </TouchableOpacity>
          </View>
        </View>
        {/* <TouchableOpacity
              onPress={() => this.props.navigation.navigate("CreateAccount")}
            >
              <Text>Create Account</Text>
            </TouchableOpacity> */}
        {/* </View> */}
      </ScrollView>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: "white",
    flex: 1,
    padding: 20,
  },

  logoContainer: {
    justifyContent: "center",
    alignItems: "center",
    paddingTop: 40,
  },

  logo: {
    flex: 1,
    width: 200,
    height: 100,
    resizeMode: "contain",
  },
  formContainer: {
    paddingTop: 40,
  },
  formTitle: {
    color: "#30694B",
    paddingBottom: 10,
  },
  input: {
    height: 50,
    backgroundColor: "white",
    borderWidth: 1,
    borderRadius: 10,
    borderColor: "#b5b5b5",
    marginBottom: 20,
    color: "#383838",
    paddingHorizontal: 10,
  },

  buttonContainer: {
    alignItems: "center",
    padding: 5,
    marginBottom: 10,
    borderRadius: 10,
    // Navy brand color
    backgroundColor: "#30694B",
  },
  buttonText: {
    borderColor: "white",
    fontSize: 15,
    color: "white",
    padding: 10,
    borderRadius: 20,
  },
  accountLink: {
    justifyContent: "center",
    alignItems: "baseline",
    paddingTop: 20,
    flexDirection: "row",
  },
});
