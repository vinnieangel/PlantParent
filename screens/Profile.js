import React, { useState, Component } from "react";
import { Modal, useWindowDimensions } from "react-native";
import { Button } from "react-native";
import {
  View,
  text,
  SafeAreaView,
  StyleSheet,
  TextInput,
  TouchableOpacity,
} from "react-native";
import {
  Avatar,
  Title,
  Caption,
  Text,
  TouchableRipple,
} from "react-native-paper";

import Ionicons from "react-native-vector-icons/Ionicons";

export default class ProfileScreen extends Component {
  userID = this.props.route.params.userID;
  state = {
    modalVisible1: false,
    modalVisible2: false,
    username: "",
    password: "",
    confirmedUsername: "",
    confirmedPassword: "",
  };

  // fetches the server and deleted the account
  async deleteAccount() {
    await fetch("https://plantparent506.herokuapp.com/users/delete", {
      method: "DELETE",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify({
        userID: this.userID,
      }),
    })
      .then(() => {
        window.alert("The account has been deleted.");
        this.props.navigation.navigate("Login");
      })
      .catch((err) => {
        console.log("Error: " + err);
      });
  }

  async updateUsername() {
    console.log(this.state.confirmedUsername);
    await fetch("https://plantparent506.herokuapp.com/users/updateUsername", {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify({
        userID: this.props.route.params.userID,
        username: this.state.confirmedUsername,
      }),
    })
      .then(() => {})
      .catch((err) => {
        console.log("Error: " + err);
      });
  }

  async updatePassword() {
    console.log(this.state.confirmedPassword);
    await fetch("https://plantparent506.herokuapp.com/users/updatePassword", {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify({
        userID: this.props.route.params.userID,
        password: this.state.confirmedPassword,
      }),
    })
      .then(() => {
        this.props.navigation.navigate("Login");
      })
      .catch((err) => {
        console.log("Error: " + err);
      });
  }

  handleUsername = (text) => {
    this.setState({ username: text });
  };

  handleConfirmedUsername = (text) => {
    this.setState({ confirmedUsername: text });
  };

  handlePassword = (text) => {
    this.setState({ password: text });
  };

  handleConfirmedPassword = (text) => {
    this.setState({ confirmedPassword: text });
  };

  checkIfEqual1() {
    if (this.state.username.length < 8) {
      this.state.username = "";
      this.state.confirmedUsername = "";
      window.alert(
        "New user name is too short. It must be at least 8 characters."
      );
    } else if (this.state.username === "") {
      this.state.confirmedUsername = "";
      window.alert("Type in the new user name you prefer!");
    } else if (this.state.confirmedUsername === "") {
      this.state.username = "";
      window.alert("Please confirm the new user name!");
    } else {
      if (this.state.username === this.state.confirmedUsername) {
        // POST to database
        this.updateUsername();
        window.alert("New user name is updated.");
      } else {
        this.state.username = "";
        this.state.confirmedUsername = "";
        window.alert("Confirming new user name failed.");
      }
    }
  }

  checkIfEqual2() {
    if (this.state.password.length < 8) {
      this.state.password = "";
      this.state.confirmedPassword = "";
      window.alert(
        "New password is too short. It must be at least 8 characters."
      );
    } else if (this.state.password === "") {
      this.state.confirmedPassword = "";
      window.alert("Type in the new password!");
    } else if (this.state.confirmedPassword === "") {
      this.state.password = "";
      window.alert("Please confirm the password!");
    } else {
      if (this.state.password === this.state.confirmedPassword) {
        // POST to database
        this.updatePassword();
        window.alert("New password is updated. Redirected to Login screen.");
      } else {
        this.state.password = "";
        this.state.confirmedPassword = "";
        window.alert("Confirming new password failed.");
      }
    }
  }
  render() {
    return (
      <SafeAreaView style={styles.container}>
        <View style={styles.userInfoSection}>
          <View style={{ flexDirection: "row", marginTop: 15 }}>
            <Avatar.Image
              source={{
                uri: "https://pngimg.com/uploads/minions/minions_PNG25.png",
              }}
              size={80}
            />
            <View style={{ marginLeft: 20 }}>
              <Title
                style={[
                  styles.title,
                  {
                    marginTop: 15,
                    marginBottom: 5,
                  },
                ]}
              >
                Name
              </Title>
              <Caption style={styles.caption}>User ID</Caption>
            </View>
          </View>
        </View>

        <View style={styles.userInfoSection}>
          <View style={styles.row}>
            <Ionicons name="flag-outline" color="#777777" size={20} />
            <Text style={{ color: "#777777", marginLeft: 20 }}>
              Current Location
            </Text>
          </View>
          <View style={styles.row}>
            <Ionicons name="mail-outline" color="#777777" size={20} />
            <Text style={{ color: "#777777", marginLeft: 20 }}>
              Email Address
            </Text>
          </View>
        </View>

        <View style={styles.settingWrapper}>
          <Modal
            animationType="slide"
            transparent={true}
            visible={this.state.modalVisible1}
          >
            <View style={styles.changeInfo}>
              <View style={styles.modalView}>
                <Text style={styles.modalTitle}> Change Username </Text>
                <Text style={styles.textInputLabel}> Username </Text>
                <TextInput
                  style={styles.textInput}
                  placeholder="New username"
                  placeholderTextColor="#b5b5b5"
                  autoCapitalize="none"
                  onChangeText={this.handleUsername}
                />
                <Text style={styles.textInputLabel}> Confirm username</Text>
                <TextInput
                  style={styles.textInput}
                  placeholder="Confirm username"
                  placeholderTextColor="#b5b5b5"
                  autoCapitalize="none"
                  onChangeText={this.handleConfirmedUsername}
                />
                <Button
                  title="Save"
                  onPress={() => {
                    this.setState({ modalVisible1: false }),
                      this.checkIfEqual1();
                  }}
                />
                <Button
                  title="Cancel"
                  onPress={() => this.setState({ modalVisible1: false })}
                />
              </View>
            </View>
          </Modal>

          <TouchableRipple
            onPress={() => {
              this.setState({ modalVisible1: true });
            }}
          >
            <View style={styles.settingItem}>
              <Ionicons name="settings-outline" color="#FF6347" size={25} />
              <Text style={styles.settingItemText}>Change Username</Text>
            </View>
          </TouchableRipple>

          <Modal
            animationType="slide"
            transparent={true}
            visible={this.state.modalVisible2}
          >
            <View style={styles.changeInfo}>
              <View style={styles.modalView}>
                <Text style={styles.modalTitle}> Change Password </Text>
                <Text style={styles.textInputLabel}> New Password </Text>
                <TextInput
                  style={styles.textInput}
                  placeholder="New Password"
                  secureTextEntry
                  placeholderTextColor="#b5b5b5"
                  autoCapitalize="none"
                  onChangeText={this.handlePassword}
                />
                <Text style={styles.textInputLabel}> Password </Text>
                <TextInput
                  style={styles.textInput}
                  placeholder="Confirm New Password"
                  secureTextEntry
                  placeholderTextColor="#b5b5b5"
                  autoCapitalize="none"
                  onChangeText={this.handleConfirmedPassword}
                />
                <Button
                  title="Save"
                  onPress={() => {
                    this.setState({ modalVisible2: false }),
                      this.checkIfEqual2();
                  }}
                />
                <Button
                  title="Cancel"
                  onPress={() => this.setState({ modalVisible2: false })}
                />
              </View>
            </View>
          </Modal>
          <TouchableRipple
            onPress={() => {
              this.setState({ modalVisible2: true });
            }}
          >
            <View style={styles.settingItem}>
              <Ionicons name="lock-closed-outline" color="#FF6347" size={25} />
              <Text style={styles.settingItemText}>Change Password</Text>
            </View>
          </TouchableRipple>
          <TouchableRipple
            onPress={() => {
              this.deleteAccount();
            }}
          >
            <View style={styles.settingItem}>
              <Ionicons name="trash-outline" color="#FF6347" size={25} />
              <Text style={styles.settingItemText}>Delete Account</Text>
            </View>
          </TouchableRipple>
        </View>
      </SafeAreaView>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "white",
  },
  userInfoSection: {
    paddingHorizontal: 30,
    marginBottom: 25,
  },
  title: {
    fontSize: 24,
    fontWeight: "bold",
  },
  caption: {
    fontSize: 14,
    lineHeight: 14,
    fontWeight: "500",
  },
  row: {
    flexDirection: "row",
    marginBottom: 10,
  },
  infoBoxWrapper: {
    borderBottomWidth: 1,
    borderTopWidth: 1,
    flexDirection: "row",
    height: 100,
  },
  infoBox: {
    width: "50%",
    alignItems: "center",
    justifyContent: "center",
  },
  settingItem: {
    flexDirection: "row",
    paddingVertical: 15,
    paddingHorizontal: 30,
  },
  settingItemText: {
    color: "#777777",
    marginLeft: 20,
    fontWeight: "600",
    fontSize: 16,
    lineHeight: 26,
  },
  input: {
    height: 50,
    backgroundColor: "white",
    borderWidth: 1,
    borderRadius: 10,
    borderColor: "#b5b5b5",
    marginBottom: 20,
    color: "#383838",
    paddingHorizontal: 10,
  },
  modalButton: {
    backgroundColor: "white",
    color: "black",
  },
  textInput: {
    borderBottomWidth: 1,
    borderBottomColor: "#000000",
  },
  changeInfo: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    marginTop: 22,
  },
  modalView: {
    margin: 20,
    backgroundColor: "white",
    borderRadius: 20,
    padding: 35,
    alignItems: "center",
    shadowColor: "#000",
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 4,
    elevation: 5,
    borderRadius: 15,
    padding: 25,
    alignItems: "flex-start",
  },
  modalTitle: {
    fontSize: 20,
    paddingBottom: 20,
    fontWeight: "bold",
  },
  textInput: {
    width: 280,
    height: 40,
    backgroundColor: "white",
    borderWidth: 0.5,
    borderRadius: 10,
    borderColor: "#b5b5b5",
    marginBottom: 20,
    color: "#383838",
    paddingHorizontal: 10,
  },
  textInputLabel: {
    paddingBottom: 10,
  },
  changeInfo: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    marginTop: 22,
  },
});
