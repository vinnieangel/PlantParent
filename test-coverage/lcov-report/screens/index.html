
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for screens</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../index.html">All files</a> screens</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">40.82% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>20/49</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">0% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>0/8</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">41.67% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>10/24</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">40.82% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>20/49</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
    </div>
    <div class='status-line low'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="CreateAccount.js"><a href="CreateAccount.js.html">CreateAccount.js</a></td>
	<td data-value="43.75" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 43%"></div><div class="cover-empty" style="width: 57%"></div></div>
	</td>
	<td data-value="43.75" class="pct low">43.75%</td>
	<td data-value="16" class="abs low">7/16</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="50" class="pct high">50%</td>
	<td data-value="8" class="abs high">4/8</td>
	<td data-value="43.75" class="pct low">43.75%</td>
	<td data-value="16" class="abs low">7/16</td>
	</tr>

<tr>
	<td class="file low" data-value="GardenArea.js"><a href="GardenArea.js.html">GardenArea.js</a></td>
	<td data-value="36.84" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 36%"></div><div class="cover-empty" style="width: 64%"></div></div>
	</td>
	<td data-value="36.84" class="pct low">36.84%</td>
	<td data-value="19" class="abs low">7/19</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="28.57" class="pct low">28.57%</td>
	<td data-value="7" class="abs low">2/7</td>
	<td data-value="36.84" class="pct low">36.84%</td>
	<td data-value="19" class="abs low">7/19</td>
	</tr>

<tr>
	<td class="file low" data-value="Login.js"><a href="Login.js.html">Login.js</a></td>
	<td data-value="42.86" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 42%"></div><div class="cover-empty" style="width: 58%"></div></div>
	</td>
	<td data-value="42.86" class="pct low">42.86%</td>
	<td data-value="14" class="abs low">6/14</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="44.44" class="pct low">44.44%</td>
	<td data-value="9" class="abs low">4/9</td>
	<td data-value="42.86" class="pct low">42.86%</td>
	<td data-value="14" class="abs low">6/14</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank">istanbul</a>
                at Thu Apr 08 2021 15:58:39 GMT-0500 (Central Daylight Time)
            </div>
        </div>
        <script src="../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../sorter.js"></script>
        <script src="../block-navigation.js"></script>
    </body>
</html>
    