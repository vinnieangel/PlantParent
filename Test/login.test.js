import React from "react";
import ReactDOM from "react-dom";
import { render, fireEvent } from "@testing-library/react-native";
import Login from "../screens/Login";
import { login } from "../screens/Login";
import "@testing-library/jest-dom/extend-expect";
import getByTestId from "@testing-library/dom";

test("render Login input component properly", () => {
  const { debug, getByTestId, getByText, getByPlaceholderText } = render(
    <Login />
  );
  const input = getByPlaceholderText("Username or email");
  expect(input).toBeDefined();
});

test("render password input component properly", () => {
  const { debug, getByTestId, getByText, getByPlaceholderText } = render(
    <Login />
  );
  const password = getByPlaceholderText("Password");
  expect(password).toBeDefined();
});

test("renders without crashing", () => {
  const { debug, getByTestId, getByText, getByPlaceholderText } = render(
    <Login />
  );
});

test("render password input component properly", () => {
  const { debug, getByTestId, getByText, getByPlaceholderText } = render(
    <Login />
  );
  const password = getByTestId("button1");
  expect(password).toBeDefined();
});

test("increments count", () => {
  const { getByTestId, getByText } = render(<Login />);
  fireEvent.press(getByTestId("button1"));
});
