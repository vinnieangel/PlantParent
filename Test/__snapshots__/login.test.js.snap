// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test of Login component 1`] = `ShallowWrapper {}`;
